x/mongo/driver/topology/server.go

```diff
@@ -239,7 +246,6 @@ func (s *Server) Connect(updateCallback updateTopologyCallback) error {
 	s.updateTopologyCallback.Store(updateCallback)
 
 	if !s.cfg.monitoringDisabled && !s.cfg.loadBalanced {
-		s.rttMonitor.connect()
```

Comment:
We should only avoid starting the RTT monitor when streaming heartbeats have been explicitly disabled because we assume they are running in a FaaS environment. We should continue using the existing behavior in all other circumstances.

```go
if !s.cfg.monitoringDisabled && !s.cfg.loadBalanced {
	// Don't start the RTT monitor if the user explicitly disabled
	// streaming heartbeats because we assume their application
	// is running in a FaaS environment, where additional database
	// connections per running instance can be extremely problematic.
	if isStreamingEnabled(s) {
		s.rttMonitor.connect()
	}
	...